creme/creme_core/views/enumerable.py

```diff
@@ -18,21 +18,44 @@
 
 from django.core.exceptions import FieldDoesNotExist
 from django.http import Http404
-from django.shortcuts import get_object_or_404
+
+from creme.creme_core.enumerators import CustomFieldEnumerator
 
 from ..core.enumerable import enumerable_registry
 from ..core.exceptions import BadRequestError, ConflictError
 from ..http import CremeJsonResponse
-from ..models import CustomField, CustomFieldEnumValue
+from ..models import CustomField
 from .generic import base
 
 
 class ChoicesView(base.ContentTypeRelatedMixin, base.CheckedView):
     response_class = CremeJsonResponse
-    field_url_kwarg = 'field'
-    registry = enumerable_registry
     limit_arg = 'limit'
     term_arg = 'term'
+    only_arg = 'only'
+
+    def get_enumerator(self):
+        raise NotImplementedError()
+
+    def get(self, request, *args, **kwargs):
+        try:
+            limit = request.GET.get(self.limit_arg)
+            limit = int(limit) if limit is not None else None
+            term = request.GET.get(self.term_arg)
+            only = request.GET.get(self.only_arg)
+            only = only.split(',') if only is not None else None
```

Comment:
Hum only  `limit = int(limit) if limit is not None else None` can raise an exception no ? If it's true, I prefer to limit the try...except... block to minimal code (in order to expose clearly what can fail).